CHT/Temperature.C

```diff
@@ -1,39 +1,64 @@
 #include "Temperature.H"
+#include "primitivePatchInterpolation.H"
+
 
 using namespace Foam;
 
 preciceAdapter::CHT::Temperature::Temperature
 (
-    const Foam::fvMesh& mesh,
-    const std::string nameT
-)
-:
-T_(
-    const_cast<volScalarField*>
-    (
-        &mesh.lookupObject<volScalarField>(nameT)
-    )
-)
+        const Foam::fvMesh& mesh,
+        const std::string nameT
+        )
+    :
+      T_(
+          const_cast<volScalarField*>
+          (
+              &mesh.lookupObject<volScalarField>(nameT)
+              )
+          ),
+
+      mesh_(mesh)
+
 {
     dataType_ = scalar;
+
 }
 
 void preciceAdapter::CHT::Temperature::write(double * buffer)
 {
+
+    //If we have a nearest projection mapping, we interpolate from the centres to the nodes
```

Comment:
This is what I meant about comments above. This comment is nice, maybe add a bit more details.